### DISCUSSION TITLE
Attaching armor

### DISCUSSION THREAD

**Discussion by 7Bpencil** [bepu/bepuphysics2#121] [Q&A]
Hello Norbo! I am thinking of ways how to do it.
There are many constraints: 
* effective thickness is determined by a ray cast
* armor meshes can be triangle-dense
* each plate has its own properties, that can affect armor performance (rolled/cast, steel/aluminum, e.t.c.)
* parts, that can act like armor, should be taken into account too (wheels, gun barrels, tracks, e.t.c.)

So it means, that I should use low-poly mesh for general collisions and a bunch of separate meshes for projectile tests
(I simulate projectiles myself via sweep tests and ray cast).
![wefwef](https://user-images.githubusercontent.com/43344410/106703813-cffc6f00-660c-11eb-8e19-ca4c21d6f121.JPG)

So how can I do it?
* I can weld each mesh to the parent body (like hull armor plates to hull low-poly mesh), and disable collision tests via SubgroupCollisionFilter. But it doesn't feel clean: each plate then should have fictional mass, and additional constraints feel unnecessary.

* I can spawn armor meshes as statics/kinematics, disable collisions, then manually update their poses every frame. Looks doable.
* Load shapes but don't spawn bodies. Then do ray test on each shape using parernt pose. Looks doable too.

Do I have other options?

**Comment by RossNordby** [comment]
The most direct option would be to create a `Compound` (or, if there are a ton of children, perhaps a `BigCompound`) that includes the new shapes.

Collision callbacks and `IShapeRayHitHandler` include which child of a compound was involved in a collision.

And, as usual, I strongly recommend using a convex decomposition of the mesh rather than the mesh directly; it'll be massively faster.
